### PR TITLE
Creation of rack using existing VPC and subnet

### PR DESCRIPTION
**Pull request by lucasmacedot**
### What is the feature/fix?

** Describe the task and what is the goal for it. **
Currently there's no way to create a rack using an existing VPC and existing subnets
So, the goal with this PR a install of a new cluster, specifying the VPC and subnets to deploy  the cluster

```
convox rack install aws convox-staging \
  vpc_id=vpc-foo123 \
  private_subnets_ids=subnet-foo123,subnet-foo123,subnet-foo123 \
  public_subnets_ids=subnet-foo123,subnet-foo123,subnet-foo123 \
  internet_gateway_id=igw-foo123 \
  region=ap-southeast-2
```

** Describe the bug and the solution. **
** Any additional information **

### Add screenshot or video (optional)

** Any screenshot or video capture using the feature **

### Does it has a breaking change?

** Describe the changes and if it has any breaking changes in any feature **

### How to use/test it?

** Describe how to test or use the feature **

### Checklist
- [ ] New coverage tests
- [ ] Unit tests passing
- [ ] E2E tests passing
- [ ] E2E downgrade/update test passing
- [x] Documentation updated
- [ ] No warnings or errors on Deepsource/Codecov

### CODE REVIEW

File: docs/reference/cli/rack.md [hunk 1]
```diff
@@ -2,8 +2,7 @@
 title: "rack"
 draft: false
 slug: rack
-url: /reference/cli/rack
----
+url: /reference/cli/rack ---
```
Comment (Twsouza) [inline docs/reference/cli/rack.md:5]: could you keep `---` in a new line plz?

**Review by Twsouza — CHANGES_REQUESTED** [review]

**Comment by rickyrattlesnake** [comment]
Hey @nightfury1204 @Twsouza , any updates on this feature going into the next release ?

**Comment by Twsouza** [comment]
@rickyrattlesnake @lucasmacedot the CI is giving an error on installation:

```
Error: Duplicate resource "null_resource" configuration

  on .terraform/modules/system/terraform/system/aws/main.tf line 82:
  82: resource "null_resource" "wait_for_cluster" {

A null_resource resource named "wait_for_cluster" was already declared at
.terraform/modules/system/terraform/system/aws/main.tf:76,1-44. Resource
names must be unique per type in each module.
```

could you guys please take a look?

**Comment by lucasmacedot** [comment]
> @rickyrattlesnake @lucasmacedot the CI is giving an error on installation:
> 
> ```
> Error: Duplicate resource "null_resource" configuration
> 
>   on .terraform/modules/system/terraform/system/aws/main.tf line 82:
>   82: resource "null_resource" "wait_for_cluster" {
> 
> A null_resource resource named "wait_for_cluster" was already declared at
> .terraform/modules/system/terraform/system/aws/main.tf:76,1-44. Resource
> names must be unique per type in each module.
> ```
> 
> could you guys please take a look?

Hello
Just fixed it

**Comment by Twsouza** [comment]
@lucasmacedot  still having error: 

```
Error: Duplicate variable declaration

  on .terraform/modules/system/terraform/system/aws/variables.tf line 196:
 196: variable "private_subnets_ids" {

A variable named "private_subnets_ids" was already declared at
.terraform/modules/system/terraform/system/aws/variables.tf:186,1-31.
Variable names must be unique within a module.


Error: Duplicate variable declaration

  on .terraform/modules/system/terraform/system/aws/variables.tf line 201:
 201: variable "public_subnets_ids" {

A variable named "public_subnets_ids" was already declared at
.terraform/modules/system/terraform/system/aws/variables.tf:191,1-30.
Variable names must be unique within a module.
```

**Comment by lucasmacedot** [comment]
> @lucasmacedot still having error:
> 
> ```
> Error: Duplicate variable declaration
> 
>   on .terraform/modules/system/terraform/system/aws/variables.tf line 196:
>  196: variable "private_subnets_ids" {
> 
> A variable named "private_subnets_ids" was already declared at
> .terraform/modules/system/terraform/system/aws/variables.tf:186,1-31.
> Variable names must be unique within a module.
> 
> 
> Error: Duplicate variable declaration
> 
>   on .terraform/modules/system/terraform/system/aws/variables.tf line 201:
>  201: variable "public_subnets_ids" {
> 
> A variable named "public_subnets_ids" was already declared at
> .terraform/modules/system/terraform/system/aws/variables.tf:191,1-30.
> Variable names must be unique within a module.
> ```

Fixed, sorry, that happened when I had to rebase staging

**Comment by RaphaelMacedonio** [comment]
Hello @Twsouza and @nightfury1204, do we need to change anything else here for this PR to be merged and then released? What are the next steps? Is the anything that @lucasmacedot and I can help with? 
thanks.

**Comment by nightfury1204** [comment]
@lucasmacedot @RaphaelMacedonio thanks I will run ci again

**Comment by nightfury1204** [comment]
Based on the hight availability  we have to consider the implementation. for example when high availability is enabled it expects 3 subnets in three az
I don't think this implementation has consider that

**Comment by lucasmacedot** [comment]
> Based on the hight availability we have to consider the implementation. for example when high availability is enabled it expects 3 subnets in three az. And there might be other edge cases I don't think this implementation has consider that

The scenario you are talkng about is that:
If I pass only 1 subnet, and HA is enabled, terraform should create 2 more subnets?

**Comment by nightfury1204** [comment]
> > Based on the hight availability we have to consider the implementation. for example when high availability is enabled it expects 3 subnets in three az. And there might be other edge cases I don't think this implementation has consider that
> 
> The scenario you are talkng about is that: If I pass only 1 subnet, and HA is enabled, terraform should create 2 more subnets?

it shouldn't create other 2 (it can but it will have complicated logic to figure out the subnet). In HA, it expects 3 subnets in 3 az, if we provide manually it should consider that and also in HA we expect 3 nat gateway. If you go through the code you will find out there are coupling logic around that

**Comment by RaphaelMacedonio** [comment]
Hey team, @lucasmacedot @rickyrattlesnake @nightfury1204 @Twsouza is there a better way on integrating this feature? Shall we pair somehow or get a quick call to discuss and understand the better way on moving with this?

**Comment by nightfury1204** [comment]
Hey @lucasmacedot @RaphaelMacedonio I still haven't got time to review it. Let me go through it. I will let you guys know
thanks

**Comment by nightfury1204** [comment]
@lucasmacedot can you squash the commits and rebase this branch with `staging`?

**Comment by lucasmacedot** [comment]
Hello @nightfury1204 When I created this PR I created it based on the main branch, but I was asked to rebase to staging, but at the time there were differences between the main branch and the staging branch, and this messed up the history of the branch, that's why there're rebase conflicts everytime there's a change on staging
I just created a new branch based on staging from the beginning, so these rebase problems don't happen anymore

It's in this PR: https://github.com/convox/convox/pull/677

**Pull request closed without merging**